_posts/2020-08-11-pull-requests-require-context.md

```diff
@@ -0,0 +1,160 @@
+giving our engineers
+[some reading about how we work with PRs during their onboarding](https://github.com/artsy/README/blob/master/playbooks/engineer-workflow.md#pull-requests).
+
+But beyond that our engineers lead by example. This article presents a handful of examples from Artsy repositories
+demonstrating how you can add context to your PRs to avoid unnecessary clarifying conversation.
+
+## Explain Your Reasoning
+
+You've been thinking a lot about the problem you're solving - probably significantly more than your reviewers.
+You'll save everyone time by describing the problem and sharing how you're thinking about it.
+
+### Define the problem and solution
+
+Why does this PR exist? Explain the problem it solves and describe your solution, as
+[Sarah](https://github.com/sweir27) does [in this PR](https://github.com/artsy/force/pull/3095). For bonus points,
+include alternative approaches you considered.
+
+As you are writing up the problem and solution, you might find that you've missed on the scope of your PR. Are
+there _many_ problems this PR is solving? Maybe this should be broken into smaller PRs. Is it hard to describe the
+problem because it requires multiple other PRs? Maybe those should be consolidated into one cohesive set of
+changes.
+
+### Explain interesting lines of code
+
+The reviewers aren't the only ones who can comment on lines of code.
+[Give them](https://github.com/artsy/emission/pull/2085#discussion_r378228269)
+[additional information](https://github.com/artsy/emission/pull/2085#discussion_r378230196)
+[about why](https://github.com/artsy/emission/pull/2085#discussion_r378231974) a particular line was written, as
+[David](https://github.com/ds300) does [in this PR](https://github.com/artsy/emission/pull/2085). Maybe you want
+feedback focused on that line or maybe the line has side-effects and implications that aren't obvious.
+
+### Give a guided tour of the changes
+
+[Devon](https://github.com/dblandin) takes the idea of adding context to individual lines to the next level
+[in this PR](https://github.com/artsy/reaction/pull/2774#pullrequestreview-288095754). He takes advantage of
+markdown to give us a virtual tour of the changes, at each stop providing helpful information and a link to the
+next change. It's like he's sitting next to you!
+
+## Show Your Work
+
+If your PR contains work that is beyond trivial, show your reviewers how you thought about the problem. Demonstrate
+the effects of the changes. Give them confidence that you've worked through this problem thoroughly, and you've
+brought receipts.
+
+### Make small, self-contained commits
+
+A good PR starts with good commits. Good commits are small, self-contained, and leave the codebase always in a
+working state. With good commits, reviewers can see exactly how you worked through the problem you were solving.
+[Here's a PR](https://github.com/artsy/convection/pull/645) from [Jon](https://github.com/jonallured) that
```

Comment:
@jonallured any objections to me referencing you & your PR here?

Reply:
No objections, thanks pal! ❤️